services/src/main/java/org/keycloak/userprofile/config/UPConfigUtils.java

```diff
@@ -192,8 +192,15 @@ private static void validateAttribute(KeycloakSession session, UPAttribute attri
             }
         }
         if (attributeConfig.getRequired() != null) {
-            validateRoles(attributeConfig.getRequired().getRoles(), "required.roles", errors, attributeName);
-            validateScopes(attributeConfig.getRequired().getScopes(), "required.scopes", attributeName, errors, session);
+            Set<String> roles = attributeConfig.getRequired().getRoles();
+            Set<String> scopes = attributeConfig.getRequired().getScopes();
+            validateRoles(roles, "required.roles", errors, attributeName);
+            validateScopes(scopes, "required.scopes", attributeName, errors, session);
+
+            // Role "user" required when some scopes are configured
+            if (scopes != null && !scopes.isEmpty() && (roles == null || !roles.contains(ROLE_USER))) {
```

Comment:
Configuration validation is nice, but will make configurations with empty roles but scopes invalid while this was the combination which worked before.
Migration instructions then need a note to update from 
"required-scopes": [] 
to 
"required-scopes": ["user]